nwbinspector/checks/nwb_containers.py

```diff
@@ -73,5 +73,5 @@ def check_empty_string_for_optional_attribute(nwb_container: NWBContainer):
     for field in fields:
```

Comment:
The logic for `optional_attrs` is a bit to unpack - is there a better variable name than `_d` for explaining what we're iterating over?

And just to confirm, `"args"` is all we need from the `__docval__` attached to the `__init__` of some class? No `"kwargs"` alternatives?

And the logic for detecting optionality is the presence of a `"default"` value, and if there is one we check that it's set to `None` instead of some pre-defined value? Just curious, how does the `None` logic equate to a file written in MATLAB? Does the attribute simply not show up attached to the instance of the class when read back in through PyNWB?

Reply:
> The logic for optional_attrs is a bit to unpack - is there a better variable name than _d for explaining what we're iterating over?

ok, I renamed `_d` to `arg`. 

> And just to confirm, "args" is all we need from the __docval__ attached to the __init__ of some class? No "kwargs" alternatives?

docval just has everything in "args". There is no "kwargs" field.

> how does the None logic equate to a file written in MATLAB 

In MatNWB, all attributes are written with key,value pairs, e.g. 

```matlab
NwbFile("session_start_time", now(), "session_description", "this is my description")
```

To omit an attribute, you just don't provide the key, value pair. Omitted values are read back as `[]`.

Reply:
Oky doky, sounds good then